MITLibraries/solenoid
# In 957 configure django auth

### What does this PR do?

Refactor the Solenoid application to use Django’s built-in authentication system to return functionality to the application.

#### Helpful background context

* To configure authentication with Django, I closely followed [this guide](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Django/Authentication).
* Users will not be prompted to change their password when logging in for the first time (I didn't want to spend too much time trying to get this to work), _however_ (and more importantly), **users can change their password after signing in.**
* When running the Django application locally, a reset link was only successfully generated if the `User.email` was provided.
* Password validators were configured for the Django app using [this guide](https://docs.djangoproject.com/en/4.2/topics/auth/passwords/#enabling-password-validation). 
   * Please refer to the doc to get a better understanding of the validators used to confirm password strength.
   * This is IS&T's [guide for strong passwords](http://kb.mit.edu/confluence/display/istcontrib/Strong+Passwords). Some of the standard validators configured for the app coincide with the requirements for IS&T, but it will not be a 1-to-1 match. However, I think it is sufficient for now. 
* To create user accounts, the following steps were taken:
   * Download Heroku CLI
   * Log into Heroku
   * Create a superuser account (i.e., admin) with the command:
      ```
      heroku run -a=mitlibraries-in-957-con-ytpnes python manage.py createsuperuser
      ```
   * To create users, [the admin panel](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Django/Authentication#creating_users_and_groups) was used.

### How can a reviewer see these changes?

Opening this PR created a review app on Heroku. 

1. Visit https://mitlibraries-in-957-con-ytpnes.herokuapp.com/. 
2. Confirm that you can log in successfully using the credentials shared with you (via LastPass).
3. Confirm that you can log out successfully.
4. Confirm that you can reset a forgotten password.
   * From the login page, select `Need help signing in?`
   * You should see/receive a link to reset your password.
   * Fill out the form with a new password.
   * Confirm that you can log in successfully using the new password. 
6. Confirm that you can change your password at anytime after you are logged in. 
   * After logging in, select `Change my password` from the top of the homepage.
   * Fill out the form with your old and new passwords.
      * **Note:** Check what happens when you provide an input that doesn't meet the password requirements (e.g., type in '<your username>', '123', '123456789').
   * Confirm that you can log in successfully using the new password. 

### Includes new or updated dependencies?
YES

### Reviewer checklist
- [ ] There are tests covering any new functionality
- [ ] The changes, if applicable, have been verified
